### ISSUE TITLE
https: Node HTTPS server has no socket timeout by default

### ISSUE THREAD

**Issue by bminer** [nodejs/node-v0.x-archive#7764]
The **HTTP** (not HTTPS) documentation reads:

```
By default, the Server's timeout value is 2 minutes, and sockets are
destroyed automatically if they time out. However, if you assign a
callback to the Server's 'timeout' event, then you are responsible
for handling socket timeouts.
```

This is not true for **HTTPS** servers.  As you can see from [this snippet](https://github.com/joyent/node/blob/v0.10.22/lib/tls.js#L1166-L1171), `tls.js` is overriding the socket timeout during the TLS handshake (setting it to `options.handshakeTimeout || (120 * 1000)`, as appropriate).  Then, after the TLS handshake, the socket timeout is disabled completely.

This behavior should be changed, or the documentation for the HTTPS server should be updated.

This particular issue has caused me massive headaches, as our Node HTTPS server has been crashing due to hanging connections (we have a lot of 3G modems connecting to our server).

As a workaround, I am using the following to restore the default 2 minute timeout:

```
server.on("secureConnection", function(clearText) {
    clearText.setTimeout(2 * 60 * 1000);
});
```

**EDIT**: I should also note that `clearText.setTimeout` is also **NOT** documented on the Node API.

**Comment by diegoperini** [comment]
This bug still exists afaik. Our iOS clients are able to hang their SSL handshakes which in the end causes infinite waits on server. We crash our processes when they occupy Node's event loop longer than our hard-coded threshold.

**Comment by rvagg** [comment]
@diegoperini what version of Node are you using?

**Comment by diegoperini** [comment]
`v5.1.0` on an Ubuntu Trusty based Docker container.

**Comment by rvagg** [comment]
@diegoperini would you mind creating a new issue in https://github.com/nodejs/node with the problem you're experiencing, and if possible, a reduced test case that can be used to reproduce the error even with specifics about clients and server environment. I'm afraid it's going to be difficult to get action on this over here, particularly given #8021 was merged and supposed to deal with this; it could be a separate bug, maybe a new one in v5.

**Comment by bminer** [comment]
@rvagg - The PR #8021 was never merged.  It's likely the same problem as before.  Someone please post the relevant `nodejs/node` issue number here for everyone's reference.
